File: scripts/convert-data.py
#!/usr/bin/env python3
import json
import sys

# 读取 words_real.ts
with open('../src/data/words_real.ts', 'r', encoding='utf-8') as f:
    content = f.read()

# 修正JSON格式
content = content.replace('"test type":', '"testType":')
content = content.replace('"sub-order":', '"subOrder":')
content = content.replace('"student No.":', '"studentNo":')
content = content.replace('备注', 'notes')

# 解析JSON
data_array = json.loads(content)

print(f"总共读取 {len(data_array)} 条数据")

# 按testType分组
test_groups = {
    'test1': [],
    'test2': [],
    'test3': [],
    'test4': []
}

for item in data_array:
    test_type = item['testType']
    if test_type == 'test 1':
        test_groups['test1'].append(item)
    elif test_type == 'test 2':
        test_groups['test2'].append(item)
    elif test_type == 'test 3':
        test_groups['test3'].append(item)
    elif test_type == 'test 4':
        test_groups['test4'].append(item)

# 输出统计
for key, items in test_groups.items():
    print(f"{key.upper()}: {len(items)} 题")

# 生成TypeScript文件
output = """// 正式测试数据 (Test Questions)
import { TrialItem } from '@/types';

"""

def convert_item(item, test_type):
    return {
        'testType': test_type,
        'order': int(item['order']),
        'condition': item['condition'],
        'word': item['word'],
        'meaning': item['meaning'],
        'sentence': item['sentence'],
        'theme': item['theme'],
        'subOrder': int(item['subOrder']),
        'notes': item['notes'],
        'isWarmup': False
    }

for test_key in ['test1', 'test2', 'test3', 'test4']:
    test_name = test_key.upper()
    trials = [convert_item(item, test_key) for item in test_groups[test_key]]
    
    output += f"// =====================================================\n"
    output += f"// {test_name} - {len(trials)} 题\n"
    output += f"// =====================================================\n"
    output += f"export const {test_name}_TRIALS: TrialItem[] = "
    output += json.dumps(trials, ensure_ascii=False, indent=2)
    output += ";\n\n"

# 写入文件
with open('../src/data/testData.ts', 'w', encoding='utf-8') as f:
    f.write(output)

print("\n✅ 数据转换完成！文件已保存到: testData.ts")

File: scripts/convertData.js
// 数据转换脚本 - 将 words_real.ts 转换为 testData.ts
const fs = require('fs');
const path = require('path');

// 读取 words_real.ts
const wordsRealPath = path.join(__dirname, '../src/data/words_real.ts');
const content = fs.readFileSync(wordsRealPath, 'utf-8');

// 解析JSON数组（words_real.ts本质上是一个JSON数组）
const rawData = eval(content);

// 分组
const test1 = [];
const test2 = [];
const test3 = [];
const test4 = [];

rawData.forEach(item => {
  const testType = item.testType.toLowerCase().replace(' ', '');
  const converted = {
    testType: testType,
    order: parseInt(item.order) || 0,
    condition: item.condition,
    word: item.word,
    meaning: item.meaning,
    sentence: item.sentence,
    theme: item.theme,
    subOrder: parseInt(item.subOrder) || 1,
    notes: item.notes || '',
    isWarmup: false
  };

  switch (testType) {
    case 'test1':
      test1.push(converted);
      break;
    case 'test2':
      test2.push(converted);
      break;
    case 'test3':
      test3.push(converted);
      break;
    case 'test4':
      test4.push(converted);
      break;
  }
});

// 生成 TypeScript 代码
const output = `// 正式测试数据 (Test Questions) - 自动生成
import { TrialItem } from '@/types';

// =====================================================
// TEST1 - ${test1.length} 题
// =====================================================
export const TEST1_TRIALS: TrialItem[] = ${JSON.stringify(test1, null, 2)};

// =====================================================
// TEST2 - ${test2.length} 题
// =====================================================
export const TEST2_TRIALS: TrialItem[] = ${JSON.stringify(test2, null, 2)};

// =====================================================
// TEST3 - ${test3.length} 题
// =====================================================
export const TEST3_TRIALS: TrialItem[] = ${JSON.stringify(test3, null, 2)};

// =====================================================
// TEST4 - ${test4.length} 题
// =====================================================
export const TEST4_TRIALS: TrialItem[] = ${JSON.stringify(test4, null, 2)};
`;

// 写入 testData.ts
const testDataPath = path.join(__dirname, '../src/data/testData.ts');
fs.writeFileSync(testDataPath, output);

console.log(`转换完成！`);
console.log(`TEST1: ${test1.length} 题`);
console.log(`TEST2: ${test2.length} 题`);
console.log(`TEST3: ${test3.length} 题`);
console.log(`TEST4: ${test4.length} 题`);
